File: backend/src/main/java/hu/szakdoga/backend/timetable/data/dto/LessonDTO.java
package hu.szakdoga.backend.timetable.data.dto;

import java.time.LocalTime;

public class LessonDTO {
    public final Long id;
    public final String day;
    public final LocalTime startTime;
    public final LocalTime endTime;
    public final String location;
    public final String type;
    public final Long subjectId;
    public final Long timetableId;
    public final Long teacherId;

    public LessonDTO(Long id, String day, LocalTime startTime, LocalTime endTime, String location, String type, Long subjectId, Long timetableId, Long teacherId) {
        this.id = id;
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;
        this.location = location;
        this.type = type;
        this.subjectId = subjectId;
        this.timetableId = timetableId;
        this.teacherId = teacherId;
    }
}

File: frontend/src/app/shared/model/timetable/dto/lesson.dto.ts
import { Time } from '@angular/common';

export class LessonDto {
    public readonly id: number | null;
    public readonly day: string;
    public readonly startTime: string;
    public readonly endTime: string;
    public readonly location: string;
    public readonly type: string;
    public readonly subjectId: number;
    public readonly timetableId: number;
    public readonly teacherId: number;

    constructor(
        day: string,
        startTime: string,
        endTime: string,
        location: string,
        type: string,
        subjectId: number,
        timetableId: number,
        teacherId: number,
        id: number | null = null
    ) {
        this.id = id;
        this.day = day;
        this.startTime = startTime;
        this.endTime = endTime;
        this.location = location;
        this.type = type;
        this.subjectId = subjectId;
        this.timetableId = timetableId;
        this.teacherId = teacherId;
    }
}
